src/test/java/TestLinks.java

```diff
@@ -0,0 +1,36 @@
+import org.apache.commons.validator.routines.UrlValidator;
+import org.junit.Test;
+import java.io.IOException;
+
+/**
+ * Created by pavel on 22/11/2017.
+ */
+public class TestLinks {
+
+    @Test
+    public void testLinksFromSinglePageClass()
+    {
+
+        try {
+            Main.main(new String[]{"http://www.tutorialspoint.com/"});
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    @Test
+    public void testLinksFromSinglePageMethod()
+    {
+        Links<String> links = new LinksFromSinglePage<String>("http://www.tutorialspoint.com/");
+
+        for (String url : links) {
+            System.out.println(url);
+            UrlValidator urlValidator = new UrlValidator();
+            assert (urlValidator.isValid(url));
```

Comment:
@PavelNejinskiy considering we have decorator `Links<String>` -> `Links<URL>`, validation is redundant.
Expected=String[] {"url1", "url2"}(instance attribute that is inited at `@Before` hook.
Actual = URLLinks<URL>(new FilterLinks(new StringLinks(url)));